packages/iov-bns/src/decode.spec.ts

```diff
@@ -66,6 +68,46 @@ describe("Decode", () => {
       },
     };
 
+    it("works for AddAddressToUsername", () => {
+      const transactionMessage: codecImpl.app.ITx = {
+        addUsernameAddressNftMsg: {
+          id: toUtf8("alice"),
+          address: toUtf8("0xAABB001122DD"),
+          chainID: toUtf8("wonderland"),
+        },
+      };
+      const parsed = parseMsg(defaultBaseTx, transactionMessage);
+      if (parsed.kind !== TransactionKind.AddAddressToUsername) {
+        throw new Error("unexpected transaction kind");
+      }
+      expect(parsed.username).toEqual("alice");
+      expect(parsed.payload.address).toEqual("0xAABB001122DD");
+      expect(parsed.payload.chainId).toEqual("wonderland");
+    });
+
+    it("works for RegisterBlockchain", () => {
+      const transactionMessage: codecImpl.app.ITx = {
+        issueBlockchainNftMsg: {
+          id: Encoding.toAscii("wonderland"),
+          owner: Encoding.fromHex("0011223344556677889900112233445566778899"),
+          approvals: undefined,
+          details: {
+            iov: {
+              codec: "wonderland_rules",
```

Comment:
these are both arbitrary strings for the backend, just used by client, right?

Reply:
right. To be honest I don't understand the semantics of it

Reply:
This should be documented, but I did request some info, so we can read this info, and then initialize a BnsConnection or LiskConnection or whatever.

The codec and options are only meant to be understood by iov-core... And if there is garbage there, we don't connect.